tt_eager/tt_dnn/op_library/topk/kernels/compute/topk.cpp

```diff
@@ -0,0 +1,170 @@
+        bool ascending = false;
+        cb_reserve_back(input_transposed_cb_index, Wt);
+        cb_reserve_back(index_transposed_cb_index, Wt);
+
+        for (uint32_t wt = 0; wt < Wt; wt+=2) {
+            acquire_dst(tt::DstMode::Half);
+            // local sort into k groups
+            cb_wait_front(input_cb_index, 2);
+            cb_wait_front(index_cb_index, 2);
+
+            unpack_reconfig_data_format_srca(input_cb_index);
+            transpose_wh_init_short(input_cb_index);
+            transpose_wh_tile(input_cb_index, wt, 0);
+            transpose_wh_tile(input_cb_index, wt+1, 1);
+
+            unpack_reconfig_data_format_srca(index_cb_index);
+            transpose_wh_init_short(index_cb_index);
+            transpose_wh_tile(index_cb_index, wt, 2);
+            transpose_wh_tile(index_cb_index, wt+1, 3);
+
+            // llk_topk_sort -> inplace
+            ckernel::topk_local_sort(0, (int) ascending, logk - 1);
+
+            // pack value tiles into cb_intermed1
+            pack_reconfig_data_format(input_transposed_cb_index);
+            pack_tile(0, input_transposed_cb_index, wt);
+            pack_tile(1, input_transposed_cb_index, wt+1);
+
+            // pack index tiles into cb_intermed2
+            pack_reconfig_data_format(index_transposed_cb_index);
+            pack_tile(2, index_transposed_cb_index, wt);
+            pack_tile(3, index_transposed_cb_index, wt+1);
+
+            cb_pop_front(input_cb_index, 2);
+            cb_pop_front(index_cb_index, 2);
+
+            release_dst(tt::DstMode::Half);
+            ascending = !ascending;
+        }
+
+        cb_push_back(input_transposed_cb_index, Wt);
+        cb_push_back(index_transposed_cb_index, Wt);
+
+        cb_wait_front(input_transposed_cb_index, Wt);
+        cb_wait_front(index_transposed_cb_index, Wt);
+
+        cb_reserve_back(input_transposed_cb_index, Wt);
+        cb_reserve_back(index_transposed_cb_index, Wt);
+
+
+        // iterative merge and rebuild on pairs of tiles
+        constexpr uint32_t num_iterations = logWt;
+        for (uint32_t m_iter = 0; m_iter < num_iterations; ++m_iter) {
+            bool a = false;
+            for (uint32_t left_ind = 0; left_ind < Wt / 2; left_ind += 1 << m_iter) {
+                acquire_dst(tt::DstMode::Half);
+
+                uint32_t right_ind = left_ind + (1 << m_iter);
+
+                // unpack values into dest
+                copy_tile_to_dst_init_short_with_dt(index_transposed_cb_index, input_transposed_cb_index);
+                copy_tile(input_transposed_cb_index, left_ind, 0);
+                copy_tile(input_transposed_cb_index, right_ind, 1);
+
+                // unpack indices into dest
+                copy_tile_to_dst_init_short_with_dt(input_transposed_cb_index, index_transposed_cb_index);
+                copy_tile(index_transposed_cb_index, left_ind, 2);
+                copy_tile(index_transposed_cb_index, right_ind, 3);
+
+                ckernel::topk_tile_init();
```

Comment:
Can this also be removed since we init at start?

Reply:
Oh lol, yeah for sure.

Reply:
Done